transport-msg.tex

```diff
@@ -598,12 +598,16 @@ \subsection{Device Operation}
 \subsubsection{Driver Notifications}
 \label{sec:Virtio Transport Options / Virtio Over Messages / Device Operation / Driver Notifications}
 
-After setting up one or more virtqueues, the driver \emph{MAY} send notifications
+After setting up one or more virtqueues, the driver \emph{MUST} send notifications
 to signal that new buffers are available for processing. The virtio-msg transport
-defines an optional message, \msgref{EVENT_AVAIL}, that the driver can use to
-notify the device of pending buffers if the device’s feature negotiation allows
-it (e.g., if a feature akin to \emph{NOTIFY_ON_AVAIL} is defined or if
-\emph{NEGOCIATION\_DATA} was agreed upon).
+defines the message, \msgref{EVENT_AVAIL}, that the driver will use to
+notify the device of pending buffers.
```

Comment:
The `EVENT _AVAIL` is not used by the driver but by the driver side bus
suggestion:
```
that the driver bside bus \enph{MAY}l use to notify the device of pending buffers.
```

Reply:
This is not correct.  The driver uses the transport specific virtqueue kick op and for virtio-msg transport that results in EVENT_AVAIL.  The driver side bus may send EVENT_AVAIL in band or convert it to an out-of-band notification, (or discard it but that is sub-par).